File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // nums     = [2, 20, 4, 10, 3, 4, 5]
        // hashset -> [2, 3, 4, 5, 10, 20]
        // 2, 3, 4, 5,
        // 20,
        // 4

        // convert to hashset
        unordered_set<int> hashset;
        for (int i = 0; i < nums.size(); i++) {
            hashset.insert(nums[i]);
        }

        // find the longest consecutive sequence
        int longest_length = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (hashset.count(nums[i] - 1)) {
                continue;
            }
            int curr_length = 1;
            int next = nums[i] + 1;
            while(hashset.count(next)) {
                curr_length++;
                next++;
            }
            longest_length = max(longest_length, curr_length);
        }
        return longest_length;
    }
};

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = nums.copy()

        for i in range(1, len(nums)):
            if i == 1:
                dp[i] = max(dp[i], dp[i-1])
            else:
                dp[i] = max(dp[i] + dp[i-2], dp[i-1])

        return dp[len(nums)-1]